docs/controls/controls.rst

```diff
@@ -0,0 +1,477 @@
+
+.. literalinclude:: src/imageviewer-all/ImageViewerWindow.qml
+    :lines: 1-5, 9-31, 40-46, 55, 59
+    
+Next, we create a new ``main.qml`` for our default style *Fusion*, i.e. the desktop version of the user interface.
+
+Here, we base the user interface around the ``ImageViewerWindow`` instead of the ``ApplicationWindow``. Then we add the platform specific parts to it, e.g. the ``MenuBar`` and ``ToolBar``. The only changes to these is that the calls to open the respective dialogs are made to the new functions instead of directly to the dialog controls.
+    
+.. literalinclude:: src/imageviewer-all/main.qml
+    :lines: 1-5, 9-45, 49
+
+Next, we have to create a mobile specific ``main.qml``. This will be based around the *Material* theme. Here, we keep the ``Drawer`` and the mobile specific toolbar. Again, the only change is how the dialogs are opened.
+
+.. literalinclude:: src/imageviewer-all/+android/main.qml
+    :lines: 1-6, 10-19, 23-26, 41-53, 55-61, 84, 88
+    
+The two ``main.qml`` files are placed in the file system as shown below. This lets the file selector that the QML engine automatically creates pick the right file. By default, the *Fusion* ``main.qml`` is loaded, unless the ``android`` selector is present. Then the *Material* ``main.qml`` is loaded instead.
+
+.. digraph:: selector_files
+
+    "Project Root" -> "main.qml\n(Fusion)"
+    "Project Root" -> "+android"
+    "+android" -> "main.qml\n(Material)"
+    
+Until now the style has been set in in ``main.cpp``. We could continue doing this and use ``#ifdef`` expressions to set different styles for different platform. Instead we will use the file selector mechanism again and set the style using a configuration file. Below, you can see the file for the *Material* style, but the *Fusion* file is equally simple.
+
+.. literalinclude:: src/imageviewer-all/+android/qtquickcontrols2.conf
+    :language: ini
+
+These changes has given us a joined codebase where all the document code is shared and only the differences in user interaction patterns differ. There are different ways to do this, e.g. keeping the document in a specific component that is included in the platform specific interfaces, or as in this example, by creating a common base that is extended by each platform. The best approach is best determined when you know how your specific code base looks and can decide how to separate the common from the unique.
+
+Native Dialogs
+--------------
+
+When using the image viewer you will notice that it uses a non-standard file selector dialog. This makes it look out of place.
+
+The ``Qt.labs.platform`` module can help us solve this. It provides QML bindings to native dialogs such as the file selector, font selector and colour selector. It also provides APIs to create system tray icons, as well as system global menus that sits on top of the screen (e.g. as in OS X). The cost of this is a dependency on the ``QtWidgets`` module.
```

Comment:
I think the widgets dependency is only for the fallback dialogs where native support is missing.